src/app/bios.rs

```diff
@@ -390,9 +391,32 @@ impl Component for BiosComponent {
         } else {
             find(FIRMWARENSH)?;
 
-            let cmd = format!("{} {} bios flash", FIRMWARENSH, FIRMWAREDIR);
+            let mut boot_options: Vec<(u16, Vec<u8>)> = vec!();
 
+            let order = get_boot_order();
+            if order.is_ok() {
+                println!("Preserving boot order");
+                for num in order.clone().unwrap() {
+                    boot_options.push((num, get_boot_item(num)?));
```

Comment:
Do we want to fail in this case?

Reply:
Probably not here. Should continue as a best-effort to restore the boot order at this point.

Reply:
Will continue as a best-effort.